953-yaml-commons/src/main/java/io/harness/plancreator/strategy/MatrixConfig.java

```
@@ -117,10 +121,18 @@ private void handleString(String key, Object value) {
     if (NGExpressionUtils.matchesPattern(GENERIC_EXPRESSIONS_PATTERN_FOR_MATRIX, value.toString())) {
       expressionAxes.put(
           key, new ExpressionAxisConfig(ParameterField.createExpressionField(true, (String) value, null, false)));
-    } else {
-      throw new InvalidYamlException(
-          String.format("Value provided for axes [%s] is string. It should either be a List or an Expression.", key));
+      return;
     }
+    try {
+      JsonNode jsonNode = YamlUtils.readTree(value.toString()).getNode().getCurrJsonNode();
+      if (jsonNode instanceof ArrayNode) {
+        handleList(key, YamlUtils.read(value.toString(), ArrayList.class));
+        return;
+      }
+    } catch (IOException e) {
```

add comment that this exception could be ignored